src/Language/PureScript/Pretty/JS.hs

```diff
@@ -182,14 +161,13 @@ conditional = mkPattern match
 accessor :: (Emit gen) => Pattern PrinterState JS (gen, JS)
 accessor = mkPattern match
   where
-  match (JSAccessor _ prop val) = Just (emit prop, val)
+  match (JSAccessor _ prop val) = Just (emit $ T.pack $ codePoints prop, val)
```

Comment:
I confirmed that my suspicions were correct. Instead of switching `JSAccessor` to use `Text`, I think we should remove it entirely. Then when the `Expr` in `JSIndexer` is a `StringLiteral` that matches ECMASript's `IdentifierName` grammar production, render it using static member access. This has the side benefit of using static member access where an unnecessary computed member access would have been generated previously. I'll open an issue and send a separate PR for it.